### PR TITLE
Retry git clone without --depth=1 if it failed

### PR DESCRIPTION
**Pull request by lukaszachy**
Some git repositories don't allow shallow cloning so try again without
limiting depth.

Note: mypy is great, I've used just run(command, env) and it found the problem :)

### CODE REVIEW

File: tmt/utils.py [hunk 1]
```diff
@@ -2451,6 +2451,34 @@ def get_distgit_handler_names() -> List[str]:
     return [i.usage_name for i in DistGitHandler.__subclasses__()]
 
 
+def git_clone(
+        url: str,
+        destination: str,
+        common: Common,
```
Comment (FrNecas) [inline tmt/utils.py:2457]: Hm, if we made this an instance method of `Common`, we could avoid passing it here, and then we could simply call `self.parent.git_clone(...)`. What do you think? (just a nitpick and a personal taste thing, feel free to ignore)

**Review by FrNecas — APPROVED** [review]
Neat, looks good to me! One small nitpick/potential improvement, but feel free to rebase to main and merge right away.

File: tmt/utils.py (hunk 1 above)
Comment (lukaszachy) [reply tmt/utils.py:2457]: Well, IMHO Common is too bloated and I hope that at some point we move run elsewhere while keeping logging usable :)
If you don't mind I'll keep as is

Comment (FrNecas) [reply tmt/utils.py:2458]: Yup, agreed, I also don't like `Common` very much, too much stuff there

**Pull request merged**